### ISSUE TITLE
API Exception(401): Invalid username or password

### ISSUE THREAD

**Issue by pigaz5201314** [smartcatai/smartcat-serge-sync-plugin#13]
serge push-ts smartcat.serge 

[root@zhulq-test data]# serge push-ts smart.serge 

### /data/smart.serge

Use of uninitialized value in regexp compilation at /usr/local/share/perl5/Serge/Sync/Plugin/TranslationService/Smartcat.pm line 64.

Pushing translation files to external translation service...

Running 'smartcat-cli push --disassemble-algorithm-name="Serge.io PO" --project-id=413212a4-0023-4e3d-8945-c4b2309c27e9 --token-id=30a76c71-xxxx-xxxx-xxxx-c8bab78abd9a --token=****** --project-workdir=/data/po/413212a4-0023-4e3d-8945-c4b2309c27e9 --log=/data/log/smartcat.log'...

--------------------
*** Printing to '/data/log/smartcat.log' log file ***
Running 'push' command for project '413212a4-0023-4e3d-8945-c4b2309c27e9' and translation files from '/data/po/413212a4-0023-4e3d-8945-c4b2309c27e9'...
Getting project '413212a4-0023-4e3d-8945-c4b2309c27e9'...
Failed to get project '413212a4-0023-4e3d-8945-c4b2309c27e9'.
Error:
  API Exception(401): Invalid username or password
   at /usr/local/share/perl5/Smartcat/App/Command/push.pm line 65.

--------------------
Exception occurred while processing configuration file: 
Exit code: -1; last error: 

Sync complete. 0 jobs in 1 configuration file were processed
1 config failed to be processed
#####
token_id and token

![image](https://user-images.githubusercontent.com/17328441/70612993-e2fef380-1c42-11ea-9ad7-e228980133cb.png)
![image](https://user-images.githubusercontent.com/17328441/70613094-0de94780-1c43-11ea-9589-503625b26aaa.png)

try to replace  token ,but it's not work.

help me . thanks
*******

**Comment by smartcatai** [comment]
Hi @pigaz5201314, thanks for your report.
Your project is from Smartcat Asia server installation and you need to setup https://ea.smartcat.ai as `base-url`. Current released version of plugin doesn't support that. Though you can try to install 
```
cpanm -i https://github.com/ta2-1/smartcat-app-perl/tarball/smartcat
smartcat-cli config --base-url=https://ea.smartcat.ai
```

**Comment by pigaz5201314** [comment]
> 您好@ pigaz5201314，感谢您的举报。
> 您的项目来自Smartcat Asia服务器安装，您需要将https://ea.smartcat.ai设置为`base-url`。当前发布的插件版本不支持该功能。虽然您可以尝试安装
> 
> ```
> cpanm -i https://github.com/ta2-1/smartcat-app-perl/tarball/smartcat
> smartcat-cli config --base-url=https://ea.smartcat.ai
> ```


I tried it, but it didn't work.
[root@lookbusy-server ~]# smartcat-cli config --base-url=https://ea.smartcat.ai --log=./log --token=111 --token-id=222
base_url=
log=./log
token=111
token_id=222

**Comment by pigaz5201314** [comment]
> > 您好@ pigaz5201314，感谢您的举报。
> > 您的项目来自Smartcat Asia服务器安装，您需要将[https://ea.smartcat.ai设置为`base-url`。当前发布的插件版本不支持该功能。虽然您可以尝试安装](https://ea.smartcat.ai%E8%AE%BE%E7%BD%AE%E4%B8%BA%60base-url%60%E3%80%82%E5%BD%93%E5%89%8D%E5%8F%91%E5%B8%83%E7%9A%84%E6%8F%92%E4%BB%B6%E7%89%88%E6%9C%AC%E4%B8%8D%E6%94%AF%E6%8C%81%E8%AF%A5%E5%8A%9F%E8%83%BD%E3%80%82%E8%99%BD%E7%84%B6%E6%82%A8%E5%8F%AF%E4%BB%A5%E5%B0%9D%E8%AF%95%E5%AE%89%E8%A3%85)
> > ```
> > cpanm -i https://github.com/ta2-1/smartcat-app-perl/tarball/smartcat
> > smartcat-cli config --base-url=https://ea.smartcat.ai
> > ```
> 
> 我尝试过，但是没有用。
> [root @ lookbusy-server〜]＃smartcat-cli config --base-url = https://ea.smartcat.ai --log =。/ log --token = 111 --token-id = 222
> base_url =
> 日志=。/ log
> 令牌= 111
> token_id = 222

it works.
/usr/local/share/perl5/Smartcat/Client/Configuration.pm
sub new {
    my ( $self, %p ) = ( shift, @_ );

    # class/static variables
    $p{http_timeout}    //= 180;
    $p{http_user_agent} //= 'Swagger-Codegen/1.0.0/perl';

    # authentication setting
    $p{api_key}        //= {};
    $p{api_key_prefix} //= {};
    $p{api_key_in}     //= {};

    # username and password for HTTP basic authentication
    $p{username} //= '';
    $p{password} //= '';

    # access token for OAuth
    $p{access_token} //= '';

    # base_url
    $p{base_url} //= 'https://ea.smartcat.ai';

    return bless \%p => $self;
}

**Comment by ta2-1** [comment]
Addressed in smartcatai/smartcat-app-perl#9 and https://github.com/smartcatai/smartcat-serge-sync-plugin/pull/15.
